src/dataset.py

```diff
@@ -98,43 +69,33 @@ def _augment_with_salary(record: Dict[str, str]) -> Dict[str, str]:
     return context
 
 
-def badge_lookup(badge: str) -> str:
-    try:
-        r = BADGE_DATASET.get(badge)
-        if not r:
-            html = "<p><b>No officer found for this badge number</b></p>"
-        else:
-            context = _augment_with_salary(r)
-            html = render_template("officer.j2", **context)
-
-    except Exception as err:
-        print(f"Error: {err}")
-        html = f"<p><b>Error:</b> {err}"
-
-    print("Final HTML:\n{}".format(html))
-    return html
-
-
-def _sort_names(record: _DataDict):
-    for r in sorted(record.values(), key=lambda x: x[C.FIRST]):
+def _sort_names(record: List[Tuple]) -> Tuple:
+    for r in sorted(record, key=lambda x: x[3]):
         yield r
 
 
-def name_lookup(name: str) -> str:
-    try:
-        records = NAME_DATASET.get(name)
-        if not records:
-            html = "<p><b>No officer found for this name</b></p>"
-        else:
-            htmls = []
-            for r in _sort_names(records):
-                context = _augment_with_salary(r)
-                htmls.append(render_template("officer.j2", **context))
-            html = "\n<br/>\n".join(htmls)
-
-    except Exception as err:
-        print(f"Error: {err}")
-        html = f"<p><b>Error:</b> {err}"
-
-    print("Final HTML:\n{}".format(html))
-    return html
+def name_lookup(last_name: Any, first_name: Any, badge: Any) -> str:
+    if last_name or first_name or badge:
```

Comment:
It might be better to invert this conditional and exit early for cleanliness rather than have most of the logic in this function in the if block here